### PR TITLE
Generate RPC command

### PR DESCRIPTION
**Pull request by erizocosmico**
Just review last commit

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -153,6 +160,95 @@ const (
...
+        rpc UserStore_UpdateUser(users.User) returns (users.UserStore_UpdateUserResponse);
+}
+```
+
+Note that protobuf does not support input or output types that are not messages or empty input/output, so instead of returning nothing in `UserStore_UpdateUser` it returns a message with no fields, and instead of receiving an integer in `GetUser`, receives a message with only one integer field.
+The last `error` type is ignored.
+
+### Generate RPC server implementation
+
+`gogo/protobuf` generates the interface you need to implement based on yout `.proto` file. The problem with that is that you actually have to implement that and maintain it. Instead, you can just generate it automatically with proteus.
```
Comment (dpordomingo) [inline README.md]: typo in `based on yout .proto`
^ you**r**

File: cli/proteus/main.go [hunk 2]
```diff
@@ -20,29 +20,60 @@ func main() {
...
+		},
+	}
+
 	app.Run(os.Args)
 }
 
-func action(c *cli.Context) error {
+type action func(c *cli.Context) error
+
+func initCmd(next action) func(c *cli.Context) error {
```
Comment (dpordomingo) [inline cli/proteus/main.go]: It could be used the type, couldn't be?
```func initCmd(next action) action {```

File: cli/proteus/main.go [hunk 3]
```diff
@@ -20,29 +20,60 @@ func main() {
...
+	return func(c *cli.Context) error {
+		if len(packages) == 0 {
+			return errors.New("no package provided, there is nothing to generate")
```
Comment (dpordomingo) [inline cli/proteus/main.go]: The error should start by lowercase?

File: cli/proteus/main.go (hunk 2 above)
Comment (erizocosmico) [inline cli/proteus/main.go]: Sadly, no. Was my first try, but `cli` package does a check at runtime to see if actions are of type `func(*cli.Context)` or `func(*cli.Context) error` and `action` is not that type (even though it's an alias. At least i don't have to write it twice ¯\_(ツ)_/¯

File: cli/proteus/main.go (hunk 3 above)
Comment (erizocosmico) [inline cli/proteus/main.go]: usually all go errors start with lowercase

File: README.md (hunk 1 above)
Comment (erizocosmico) [inline README.md]: oops

**Comment by erizocosmico** [comment]
Fixed @dpordomingo

**Pull request merged**